plugins/org.eclipse.fordiac.ide.structuredtextalgorithm/src/org/eclipse/fordiac/ide/structuredtextalgorithm/validation/STAlgorithmValidator.java

```diff
@@ -152,28 +152,29 @@ public void checkUniquenessOfSTAlgorithmSourceElementNamesAndFunctionNames(
 		}
 	}
 
-	@Check
-	public void checkAlgorithmForInputEvent(final STAlgorithmSource source) {
-		if (source.eResource() instanceof final STAlgorithmResource resource
-				&& resource.getInternalLibraryElement() instanceof final SimpleFBType simpleFBType) {
-			simpleFBType.getInterfaceList().getEventInputs().stream()
-					.filter(event -> simpleFBType.getAlgorithm().stream()
-							.noneMatch(alg -> alg.getName().equalsIgnoreCase(event.getName()))
-							&& source.getElements().stream().filter(STAlgorithm.class::isInstance)
-									.noneMatch(alg -> alg.getName().equalsIgnoreCase(event.getName())))
-					.forEach(event -> acceptError(
-							MessageFormat.format(Messages.STAlgorithmValidator_NoAlgorithmForInputEvent,
-									event.getName()),
-							source, 0, 0, STAlgorithmValidator.NO_ALGORITHM_FOR_INPUT_EVENT, event.getName()));
-		}
-	}
+	// TODO: change check to InputEvent 1:1 SimpleECState and move it somewhere else
+//	@Check
+//	public void checkAlgorithmForInputEvent(final STAlgorithmSource source) {
+//		if (source.eResource() instanceof final STAlgorithmResource resource
+//				&& resource.getInternalLibraryElement() instanceof final SimpleFBType simpleFBType) {
+//			simpleFBType.getInterfaceList().getEventInputs().stream()
+//					.filter(event -> simpleFBType.getAlgorithm().stream()
+//							.noneMatch(alg -> alg.getName().equalsIgnoreCase(event.getName()))
+//							&& source.getElements().stream().filter(STAlgorithm.class::isInstance)
+//									.noneMatch(alg -> alg.getName().equalsIgnoreCase(event.getName())))
+//					.forEach(event -> acceptError(
+//							MessageFormat.format(Messages.STAlgorithmValidator_NoAlgorithmForInputEvent,
+//									event.getName()),
+//							source, 0, 0, STAlgorithmValidator.NO_ALGORITHM_FOR_INPUT_EVENT, event.getName()));
+//		}
+//	}
```

Comment:
Please also adapt the `checkAlgorithmForInputEvent` to check if there is a `SimpleECAction` with an algorithm that does not exist.

Reply:
updated the check